text/0017-shared-texture.md

```diff
@@ -0,0 +1,491 @@
+    * `offset` number - The strides and offsets in bytes to be used when accessing the buffers via a memory mapping. One per plane per entry.
+    * `size` number - Size in bytes of the plane. This is necessary to map the buffers.
+    * `fd` number - File descriptor for the underlying memory object (usually dmabuf).
+  * `modifier` string _Linux_ - The modifier is retrieved from GBM library and passed to EGL driver.
+  * `supportsZeroCopyWebGpuImport` boolean _Linux_ - Indicates whether supports zero copy import to WebGPU.
+
+## Guide-level explanation
+
+### Managed API Example Usage Walkthrough
+
+In this section, we will use the output of offscreen rendering as an example of an external shared texture, importing and rendering it in another window. We use managed version of the API for easier lifetime management. 
+
+1. Import the `sharedTexture` module.
+
+```js
+const { sharedTexture } = require('electron');
+```
+
+2. Create a `BrowserWindow` and enable the offscreen rendering feature, setting the target window size.
+
+```js
+const osr = new BrowserWindow({
+  show: true,
+  webPreferences: {
+    offscreen: {
+      useSharedTexture: true
+    }
+  }
+});
+
+// It is recommended to use `setSize`, as setting width and height above in BrowserWindow may be constrained by the screen size.
+osr.setSize(3840, 2160)
+```
+
+3. Listen for the `paint` event, import the shared texture, and get a transfer object to pass to the renderer process.
+
+The `handle` is a [SharedTextureHandle](#sharedtexturehandle-object), which contains platform-specific information about the native handle.
+
+```js
+osr.webContents.on('paint', async (event) => {
+  const texture = event.texture;
+
+  if (!texture) {
+    return;
+  }
+
+  // Import the external shared texture
+  // `textureInfo` has same properties as `importSharedTexture` required,
+  // for importing your own native shared texture, fill corresponding 
+  // properties by yourself.
+  const imported = sharedTexture.importSharedTexture({
+      textureInfo: texture.textureInfo,
+      // This will be called when all references of the imported shared
+      // texture are released. When this get called, you can safely release
+      // the source native texture.
+      allReferenceReleased() {
+        // All reference are released, release the source texture.
+        texture.release()
+      }
+  });
+
+  // Use util method to transfer to renderer process
+  // This has a timeout to prevent your renderer is dead or did not
+  // register receiver with `sharedTexture.receiveFromMain`.
+  await sharedTexture.sendToRenderer(win.webContents, imported);
+
+  // You can release here as we're done using it at main process. 
+  // The receiver will need to call `release()` at the end of their usage.
+  // The underlying resource will keep alive until all holders called
+  // `release()`, and they are managed automatically to see if all
+  // holders are released.
+  imported.release();
```

Comment:
It seems like `imported` would automatically get garbage collected at the end of this function, unless a reference is retained by `sendToRenderer`. Do we need an explicit `imported.release()` call?

I vaguely remember a discussion about this in the past, so apologies if you already answered this 😅